### PR TITLE
Add builtin::export_lexically

### PR DESCRIPTION
**Pull request by leonerd**
Adds a `builtin::` function for performing the lexical-export behaviour, so that new code can do the same trickery that core `builtin::` itself achieves.

### CODE REVIEW

File: lib/builtin.t
Comment (ilmari) [inline lib/builtin.t]: It does.

File: pod/perldiag.pod
Comment (ilmari) [inline pod/perldiag.pod]: Other occurrences of `%s reference` in `perldiag` leave the article as part of the `%s`.

File: pod/perldiag.pod
Comment (ilmari) [inline pod/perldiag.pod]: The general style is to first explain what you did wrong, then explain why. Maybe something like:

> The type of a reference given to L<builtin/lexically_export> did not match the sigil of the preceding name.

File: pod/perldiag.pod
Comment (ilmari) [inline pod/perldiag.pod]: Same here. How about something like this?

> (F) L<builtin/lexically_export> was called at runtime. Because it creates new names in the lexical scope
> currently being compiled, it can only be called from a C\<BEGIN\> block in that scope.

**Review by ilmari — CHANGES_REQUESTED** [review]

**Comment by leonerd** [comment]
Thanks @ilmari. Comments addressed and force-pushed.

**Review by ilmari — APPROVED** [review]
LGTM

**Comment by tonycoz** [comment]
#19857 is still a problem here:
```
tony@venus:.../git/perl5$ ./perl -Ilib -Mbuiltin=lexically_export -e 'BEGIN { lexically_export abc => sub {}; } BEGIN { eval "abc()" }'
Built-in function 'builtin::lexically_export' is experimental at -e line 1.
perl: op.c:13404: Perl_find_lexical_cv: Assertion `PARENT_PAD_INDEX(name)' failed.
Aborted
```

**Comment by ilmari** [comment]
> #19857 is still a problem here:

Seeing as that's a problem with lexical subs in general, not specifically with exporting them, I don't think that should block this.

**Comment by leonerd** [comment]
P5P requests that this be expanded into a full RFC. I am now in the process of writing that. Until that's agreed, we should not merge this (but it can be referenced as a prototype implementation)

**Comment by leonerd** [comment]
RFC now written and PR raised: https://github.com/Perl/RFCs/pull/20

**Pull request merged**
